testlapack/dtrevc3.go

```diff
@@ -0,0 +1,209 @@
+		}
+	}
+
+	work := make([]float64, max(1, 3*n))
+	if optwork {
+		impl.Dtrevc3(side, howmny, nil, n, nil, 1, nil, 1, nil, 1, mWant, work, -1)
+		work = make([]float64, int(work[0]))
+	}
+
+	m := impl.Dtrevc3(side, howmny, selected, n, tmat.Data, tmat.Stride,
+		vl.Data, vl.Stride, vr.Data, vr.Stride, mWant, work, len(work))
+
+	prefix := fmt.Sprintf("Case side=%v, howmny=%v, n=%v, extra=%v, optwk=%v",
+		side, howmny, n, extra, optwork)
+
+	if !generalOutsideAllNaN(tmat) {
+		t.Errorf("%v: out-of-range write to T", prefix)
+	}
+	if !generalOutsideAllNaN(vl) {
+		t.Errorf("%v: out-of-range write to VL", prefix)
+	}
+	if !generalOutsideAllNaN(vr) {
+		t.Errorf("%v: out-of-range write to VR", prefix)
+	}
+
+	if m != mWant {
+		t.Errorf("%v: unexpected value of m. Want %v, got %v", prefix, mWant, m)
+	}
+
+	if howmny == lapack.SelectedEigVec {
+		for i := range selected {
+			if selected[i] != selectedWant[i] {
+				t.Errorf("%v: unexpected selected[%v]", prefix, i)
+			}
+		}
+	}
+
+	// Check that the columns of VR and VL are actually eigenvectors and
+	// that the magnitude of their largest element is 1.
+	var k int
+	for j := 0; j < n; {
+		re := tmat.Data[j*tmat.Stride+j]
+		if j == n-1 || tmat.Data[(j+1)*tmat.Stride+j] == 0 {
+			if howmny == lapack.SelectedEigVec && !selected[j] {
+				j++
+				continue
+			}
+			if right {
+				ev := blas64.Vector{vr.Stride, vr.Data[k:]}
+				var evmax float64
+				for i := 0; i < n; i++ {
+					evmax = math.Max(evmax, math.Abs(ev.Data[i*ev.Inc]))
+				}
+				if math.Abs(evmax-1) > tol {
+					t.Errorf("%v: magnitude of largest element of VR[:,%v] not 1", prefix, k)
+				}
+				if !isRightEigenvectorOf(tmat, ev, blas64.Vector{}, complex(re, 0), tol) {
+					t.Errorf("%v: column %v of VR is not real right eigenvector", prefix, k)
+				}
+			}
+			if left {
+				ev := blas64.Vector{vl.Stride, vl.Data[k:]}
+				var evmax float64
+				for i := 0; i < n; i++ {
+					evmax = math.Max(evmax, math.Abs(ev.Data[i*ev.Inc]))
+				}
+				if math.Abs(evmax-1) > tol {
+					t.Errorf("%v: magnitude of largest element of VL[:,%v] not 1", prefix, k)
+				}
+				if !isLeftEigenvectorOf(tmat, ev, blas64.Vector{}, complex(re, 0), tol) {
+					t.Errorf("%v: column %v of VL is not real left eigenvector", prefix, k)
+				}
+			}
+			k++
+			j++
+			continue
+		}
+		if howmny == lapack.SelectedEigVec && !selected[j] {
+			j += 2
+			continue
+		}
+		im := math.Sqrt(math.Abs(tmat.Data[(j+1)*tmat.Stride+j])) *
+			math.Sqrt(math.Abs(tmat.Data[j*tmat.Stride+j+1]))
+		if right {
+			evre := blas64.Vector{vr.Stride, vr.Data[k:]}
+			evim := blas64.Vector{vr.Stride, vr.Data[k+1:]}
+			var evmax float64
+			for i := 0; i < n; i++ {
+				evmax = math.Max(evmax, math.Abs(evre.Data[i*evre.Inc])+
+					math.Abs(evim.Data[i*evim.Inc]))
+			}
+			if math.Abs(evmax-1) > tol {
+				t.Errorf("%v: magnitude of largest element of VR[:,%v] not 1", prefix, k)
+			}
+			if !isRightEigenvectorOf(tmat, evre, evim, complex(re, im), tol) {
```

Comment:
Oh, I'm forgetting to check the other eigenvector `evre - i*evim` corresponding to `re-i*im`. Will fix.